scripts/opt-alive.sh.in

```diff
@@ -14,6 +14,15 @@ PASSES="-passes= argpromotion deadargelim globalopt hotcoldsplit inline ipconstp
 
 TV="-tv"
 for arg in $@; do
+  # update the -passes argument only.
+  shift
+  if [[ $arg == "-passes="* ]]; then
+    passes=${arg#-passes=}
+    passes_new=`@CMAKE_BINARY_DIR@/../scripts/rewritepass.py @LLVM_BINARY_DIR@/bin/opt $passes`
+    set -- "$@" "-passes=module(tv),$passes_new,module(tv)"
+  else
+    set -- "$@" "$arg"
```

Comment:
I think the effect of this is to keep just the last argument as set replaces $@ instead of appending, or not?

Reply:
set replaces the argument list with "$@" ".."
The `shift` above polls the first argument, so `shift`+`set` moves the first argument to the last position.
After one iteration, the arguments are placed at the original location, with the `-passes=..` argument updated only.